Heroic: fix helper programs

aidalgol opened the pull request:
- Package missing Windows helper executables for Heroic Games Launcher.
- Pin versions of helper programs where there is a newer version upstream.

Fixes #431569
Closes #434055

## Things done
- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

aidalgol committed:
heroic-unwrapped: correct meta.changelog

aidalgol committed:
heroic-unwrapped: Add missing dependencies

- comet-gog.dummy-service
- epic-integration

keenanweaver commented on pkgs/by-name/he/heroic-epic-integration/package.nix:
```diff
@@ -0,0 +1,46 @@
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Wrapper process for games launched through Heroic Games Launcher";
+    longDescription = ''
+      This is a Windows executable that pretends to be EpicGamesLauncher.exe for
+      games that expect it to be their parent process.
+    '';
+    homepage = "https://github.com/Etaash-mathamsetty/heroic-epic-integration";
+    changelog = "https://github.com/Etaash-mathamsetty/heroic-epic-integration/releases/tag/v${finalAttrs.version}";
+    license = licenses.mit;
+    maintainers = with maintainers; [ aidalgol ];
+  };
```
```suggestion
  meta = {
    description = "Wrapper process for games launched through Heroic Games Launcher";
    longDescription = ''
      This is a Windows executable that pretends to be EpicGamesLauncher.exe for
      games that expect it to be their parent process.
    '';
    homepage = "https://github.com/Etaash-mathamsetty/heroic-epic-integration";
    changelog = "https://github.com/Etaash-mathamsetty/heroic-epic-integration/releases/tag/v${finalAttrs.version}";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ aidalgol ];
  };
```

keenanweaver commented on pkgs/by-name/he/heroic-epic-integration/package.nix:
```diff
@@ -0,0 +1,46 @@
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "heroic-epic-integration";
+  version = "0.3";
```
```suggestion
  version = "0.4";
```

aidalgol replied:
Oh, oops...

keenanweaver commented on pkgs/by-name/he/heroic-epic-integration/package.nix:
```diff
@@ -0,0 +1,46 @@
+  src = fetchFromGitHub {
+    owner = "Etaash-mathamsetty";
+    repo = "heroic-epic-integration";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-Zn0MsaQd8Ro6eu8IQkMcLNGLVTUukwajkn8PRLfB+Yw=";
```
```suggestion
    tag = "v${finalAttrs.version}";
    hash = "sha256-pRgs1w4bzm5Ao0zXfaNxBAR8+h7w4I+C+bm4nT7kIgU=";
```

aidalgol replied:
Is there a difference between `rev` and `tag` on this fetcher?  I've seen a mix used throughout nixpkgs.

keenanweaver replied:
Honestly don't know. I think it's preferred to use tag when available at this point.

keenanweaver commented on pkgs/by-name/he/heroic-epic-integration/package.nix:
```diff
@@ -0,0 +1,46 @@
+    pkgsCross.mingwW64.buildPackages.gcc
+  ];
+
+  cmakeFlags = [ "-DCMAKE_TOOLCHAIN_FILE=../windows.cmake" ];
```
```suggestion
  cmakeFlags = [ (lib.cmakeFeature "CMAKE_TOOLCHAIN_FILE" "../windows.cmake") ];
```

keenanweaver commented on pkgs/by-name/ga/galaxy-dummy-service/package.nix:
```diff
@@ -0,0 +1,44 @@
+  pkgsCross,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "GalaxyCommunication-dummy";
```
```suggestion
  pname = "galaxycommunication-dummy";
```
I think either this or `galaxy-dummy-service` works here.

keenanweaver commented on pkgs/by-name/ga/galaxy-dummy-service/package.nix:
```diff
@@ -0,0 +1,44 @@
+    inherit (comet-gog.meta) changelog homepage license;
+  }
+  // {
+    description = "A dummy Windows service for Galaxy64.dll";
```
```suggestion
    description = "Dummy Windows service for Galaxy64.dll";
```
Description can't start with an article.

keenanweaver reviewed (changes requested):
Looks good! Did not test actual functionality. Not sure how to actually test these features without downloading GTA5. Added some suggestions.

## `nixpkgs-review` result for [#432183](https://github.com/NixOS/nixpkgs/pull/432183)

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 432183`

Logs: https://github.com/keenanweaver/nixpkgs-review-gha/actions/runs/16851392980


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>galaxy-dummy-service</li>
    <li>heroic</li>
    <li>heroic-epic-integration</li>
    <li>heroic-unwrapped</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>galaxy-dummy-service</li>
    <li>heroic-epic-integration</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>galaxy-dummy-service</li>
    <li>heroic-epic-integration</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>galaxy-dummy-service</li>
    <li>heroic-epic-integration</li>
  </ul>
</details>

keenanweaver reviewed (approved):
## `nixpkgs-review` result for [#432183](https://github.com/NixOS/nixpkgs/pull/432183)

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 432183`

Logs: https://github.com/keenanweaver/nixpkgs-review-gha/actions/runs/16854007428


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>galaxy-dummy-service</li>
    <li>heroic</li>
    <li>heroic-epic-integration</li>
    <li>heroic-unwrapped</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>galaxy-dummy-service</li>
    <li>heroic-epic-integration</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>galaxy-dummy-service</li>
    <li>heroic-epic-integration</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>galaxy-dummy-service</li>
    <li>heroic-epic-integration</li>
  </ul>
</details>

CommandMC commented on pkgs/by-name/he/heroic-unwrapped/package.nix:
```diff
@@ -15,6 +15,8 @@
   nile,
   comet-gog,
+  galaxy-dummy-service,
+  heroic-epic-integration,
```
Note: Heroic (at least the latest release) depends on v0.3, seems like this would make it use v0.4

keenanweaver replied:
Did not realize that. Here's a PR that bumps it to 0.4. I think reverting it back to 0.3 is best. https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/pull/4863

CommandMC replied:
Ideally the versions of all helper binaries would be pinned here (I'm not a NixOS user, so I'm not aware of how you guys handle past package versions). If we want to update one of the helper binaries used in Heroic, we
1. publish a new release in the relevant repo
2. adjust a script in Heroic's repository to download the binaries from the new release
3. ship a new Heroic version, whenever that's ready, with the new binaries, while the old already-released version continues to use the versions it came with

Basically every release of our helper binaries contains breaking changes, or at least isn't tested for functionality with the last release of Heroic (because that's not what it's intended for). Meaning not pinning the version here might cause issues in the future
To give you an example, I'm currently working on more intuitive cloud save support, which requires changes for Legendary and gogdl. If a version with those changes is used by the latest release of Heroic, save sync will not function at all

It's fine by me if this doesn't happen *in this PR*, but I'd certainly recommend implementing this at some point

keenanweaver reviewed (commented):
Everything functioning correctly as far as I can tell.

What's the maintenance you have in mind for all these different package versions? I can see things getting out of hand quickly with all the updates done to these components.

A bit out of my wheelhouse on this. Trying to understand if this is the best approach for the Heroic package. It just feels like there might be an easier way to do this. I looked around nixpkgs to see other examples of 'latest' vs. 'versioned' components. There are packages like `podofo` and `uhttpmock` which have distinct packages per version. In `openrct2`, the dependencies are built into the same derivation (though those are just `fetchurl`). Would it be desirable to do that for Heroic? Probably need someone more experienced than me to chime in on this.

## `nixpkgs-review` result for [#432183](https://github.com/NixOS/nixpkgs/pull/432183)

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 432183`

Logs: https://github.com/keenanweaver/nixpkgs-review-gha/actions/runs/16907509795


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>comet-gog</li>
    <li>comet-gog_0_2_0</li>
    <li>galaxy-dummy-service_0_2_0</li>
    <li>heroic</li>
    <li>heroic-epic-integration_0_3</li>
    <li>heroic-unwrapped</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>comet-gog</li>
    <li>comet-gog_0_2_0</li>
    <li>galaxy-dummy-service_0_2_0</li>
    <li>heroic-epic-integration_0_3</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>comet-gog</li>
    <li>comet-gog_0_2_0</li>
    <li>galaxy-dummy-service_0_2_0</li>
    <li>heroic-epic-integration_0_3</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>comet-gog</li>
    <li>comet-gog_0_2_0</li>
    <li>galaxy-dummy-service_0_2_0</li>
    <li>heroic-epic-integration_0_3</li>
  </ul>
</details>

CommandMC commented:
This is likely not how you (intend to) do things in Nix land, but it'd be possible to just keep the binaries Heroic ships instead of building them yourself. FWIW this is how basically every other Linux package of Heroic works

aidalgol commented:
> Everything functioning correctly as far as I can tell.
> 
> What's the maintenance you have in mind for all these different package versions? I can see things getting out of hand quickly with all the updates done to these components.

My plan is to manually bump them when Heroic releases come out, and see how that goes.

> A bit out of my wheelhouse on this. Trying to understand if this is the best approach for the Heroic package. It just feels like there might be an easier way to do this. I looked around nixpkgs to see other examples of 'latest' vs. 'versioned' components. There are packages like `podofo` and `uhttpmock` which have distinct packages per version. In `openrct2`, the dependencies are built into the same derivation (though those are just `fetchurl`).

I think embedded derivations is generally discouraged.  It would also make it difficult (or at least awkward) to override any of the package versions, and these packages can potentially be used outside of Heroic.

TomaSajt commented on pkgs/by-name/co/comet-gog/package.nix:
```diff
@@ -5,33 +5,31 @@
 
   cargoHash = "sha256-K0lQuk2PBwnVlkRpYNo4Z7to/Lx2fY6RIlkgmMjvEtc=";
 
-  # error: linker `aarch64-linux-gnu-gcc` not found
-  postPatch = ''
-    rm .cargo/config.toml
-  '';
```
Have you tried building on `aarch64-linux`?
This was needed for it.

(Not sure why nixpkgs review isn't building this package on non-`x86_64-linux` systems)

aidalgol replied:
No, I'm not set up to build nixpkgs for other architectures.  It wasn't clear from the comment whether this error was happening on all architectures.  Can we make it an `optionalString` on whether the platform is `aarch64-linux`?

TomaSajt replied:
All packages do this unconditionally currently in nixpkgs.

One package has this comment:
```
    # Upstream started to redirect aarch64-unknown-linux-gnu to aarch64-linux-gnu-gcc
    # for their CI which breaks compiling on aarch64 in nixpkgs:
    #  error: linker `aarch64-linux-gnu-gcc` not found
```

If you want to, you can adapt this somehow.

aidalgol replied:
Reverted.

TomaSajt commented on pkgs/by-name/ga/galaxy-dummy-service_0_2_0/package.nix:
```diff
@@ -0,0 +1,47 @@
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkgsCross.mingwW64.buildPackages.gcc
```
Is this using `pkgsCross` (& friends) allowed in nixpkgs? I don't remember the policy...

TomaSajt replied:
You might be able to find an answer if you ask in the Nix Cross matrix channel

TomaSajt replied:
Alright

aidalgol commented on pkgs/by-name/co/comet-gog_0_2_0/package.nix:
```diff
@@ -0,0 +1,24 @@
+}:
+
+rustPlatform.buildRustPackage (finalAttrs: {
+  inherit (comet-gog) pname;
```
I'd rather keep this `inherit`, since we still want to reuse `comet-gog.meta` further down the file.

TomaSajt reviewed (commented):
Is locking down the dependency versions really necessary? What fails if we don't pin them?

aidalgol commented:
> Is locking down the dependency versions really necessary? What fails if we don't pin them?

See https://github.com/NixOS/nixpkgs/pull/432183#discussion_r2266370149

TomaSajt commented:
Ah, I see.

In any case, I'll open a PR that handles the pinning of `comet-gog` in a more "ergonomic" way. (aka. how I'd like it to be).
You can take inspiration, if you want to.

I'll also include the dummy service thing as a passthru, so you'll be able to do `comet-gog.dummy-service` for it.

TomaSajt committed:
comet-gog_heroic: init at 0.2.0; comet-gog: refactor

TomaSajt committed:
comet-gog: add dummy-service passthru

TomaSajt committed:
comet-gog: add aidalgol to maintainers

SuperSandro2000 commented on pkgs/by-name/ga/galaxy-dummy-service/package.nix:
```diff
@@ -0,0 +1,44 @@
+    runHook preInstall
+
+    mkdir $out
+    cp GalaxyCommunication.exe $out/
```
Shouldn't this be under $out/bin?

aidalgol committed:
heroic-unwrapped: remove update script

We need to keep the helper programs at the same versions as upstream, so
updating just this package is insufficient.

aidalgol committed:
legendary-heroic: move beside heroic-unwrapped

This is a fork only for Heroic, so move the nix file for this derivation
to beside heroic-unwrapped, but expose it as a passthru for ease of
overriding.

aidalgol committed:
heroic-unwrapped: use correct comet-gog version

aidalgol commented:
Cherry-picked all commits from #434055 and rebased this PR.

keenanweaver commented:
Getting an error in the logs, but the multiplayer component is still working. (As in, I can see servers and join)
```
(14:47:48) [INFO]:    [Backend]:         Launching Heretic + Hexen (1776058590)
(14:47:48) [INFO]:    [Backend]:         Checking if wine version exists: Proton - proton-ge-bin
(14:47:48) [INFO]:    [Backend]:         Checking if wine version exists: Proton - proton-ge-bin
(14:47:48) [INFO]:    [Backend]:         Checking if wine version exists: Proton - proton-ge-bin
(14:47:48) [INFO]:    [Backend]:         Checking if wine version exists: Proton - proton-ge-bin
(14:47:48) [DEBUG]:   [Backend]:         Running Wine command: createprefix
(14:47:48) [INFO]:    [Backend]:         Preventing machine to sleep
(14:47:50) [INFO]:    [Backend]:         Checking if wine version exists: Proton - proton-ge-bin
(14:47:50) [DEBUG]:   [Backend]:         Running Wine command: cmd /c winepath -u C:\ProgramData\GOG.com\Galaxy\redists\GalaxyCommunication.exe
(14:47:52) [ERROR]:   Failed to install GalaxyCommunication dummy into the prefix: Error: ENOENT, build/bin/x64/win32/GalaxyCommunication.exe not found in /nix/store/g87rxkfkfs11r0w7wmsrcdz9hbw1f6nw-heroic-unwrapped-2.18.1/opt/heroic/resources/app.asar
    at createError (node:electron/js2c/node_init:2:2060)
    at Object.copyFile (node:electron/js2c/node_init:2:4140)
    at Ts (/nix/store/g87rxkfkfs11r0w7wmsrcdz9hbw1f6nw-heroic-unwrapped-2.18.1/opt/heroic/resources/app.asar/build/main/main.js:142:932)
    at Module.Qc [as launch] (/nix/store/g87rxkfkfs11r0w7wmsrcdz9hbw1f6nw-heroic-unwrapped-2.18.1/opt/heroic/resources/app.asar/build/main/main.js:16:5680)
    at No (/nix/store/g87rxkfkfs11r0w7wmsrcdz9hbw1f6nw-heroic-unwrapped-2.18.1/opt/heroic/resources/app.asar/build/main/main.js:128:1469)
    at Session.<anonymous> (node:electron/js2c/browser_init:2:106823)
(14:47:53) [INFO]:    [Gog]:             Launching Comet!
(14:47:53) [INFO]:    [Gog]:             Launching Heretic + Hexen: HEROIC_APP_NAME=1776058590 HEROIC_APP_RUNNER=gog GAMEID=umu-0 HEROIC_APP_SOURCE=gog STORE=gog STEAM_COMPAT_INSTALL_PATH="/home/keenan/Games/Heroic/Heretic + Hexen" PULSE_SINK=Game LD_PRELOAD= STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/keenan/.local/share/.steam/steam WINEPREFIX="/home/keenan/Games/Heroic/Prefixes/default/Heretic  Hexen" STEAM_COMPAT_DATA_PATH="/home/keenan/Games/Heroic/Prefixes/default/Heretic  Hexen" PROTONPATH=/home/keenan/.config/heroic/tools/proton/proton-ge-bin WINE_FULLSCREEN_FSR=0 PROTON_ENABLE_WAYLAND=1 PROTON_ENABLE_HDR=1 PROTON_USE_WOW64=1 PROTON_DISABLE_NVAPI=1 PROTON_EAC_RUNTIME=/home/keenan/.config/heroic/tools/runtimes/eac_runtime PROTON_BATTLEYE_RUNTIME=/home/keenan/.config/heroic/tools/runtimes/battleye_runtime STEAM_COMPAT_APP_ID=0 SteamAppId=0 SteamGameId="heroic-Heretic + Hexen" PROTON_LOG_DIR=/home/keenan/.local/share GOGDL_CONFIG_PATH=/home/keenan/.config/heroic/gogdlConfig /nix/store/g87rxkfkfs11r0w7wmsrcdz9hbw1f6nw-heroic-unwrapped-2.18.1/opt/heroic/resources/app.asar.unpacked/build/bin/x64/linux/gogdl --auth-config-path /home/keenan/.config/heroic/gog_store/auth.json launch "/home/keenan/Games/Heroic/Heretic + Hexen" 1776058590 --no-wine --wrapper "obs-gamecapture /etc/profiles/per-user/keenan/bin/mangohud --dlsym "/usr/bin/umu-run"" --platform windows
(14:47:53) [DEBUG]:   [Gog]:             Logging to file(s) "/home/keenan/.local/state/Heroic/logs/games/1776058590_gog/launch.log", "/home/keenan/.local/state/Heroic/logs/runners/gog.log"
(14:47:53) [INFO]:    [Backend]:         Preventing display to sleep
(14:48:37) [INFO]:    [Gog]:             Killing Comet!
```

When installing a new game. Again, multiplayer is seemingly working as intended.

```
(14:54:24) [DEBUG]:   [Backend]:         Running Wine command: cmd /c winepath -u C:\ProgramData\GOG.com\Galaxy\redists\GalaxyCommunication.exe
(14:54:26) [ERROR]:   Failed to install GalaxyCommunication dummy into the prefix: Error: ENOENT, build/bin/x64/win32/GalaxyCommunication.exe not found in /nix/store/g87rxkfkfs11r0w7wmsrcdz9hbw1f6nw-heroic-unwrapped-2.18.1/opt/heroic/resources/app.asar
    at createError (node:electron/js2c/node_init:2:2060)
    at Object.copyFile (node:electron/js2c/node_init:2:4140)
    at Ts (/nix/store/g87rxkfkfs11r0w7wmsrcdz9hbw1f6nw-heroic-unwrapped-2.18.1/opt/heroic/resources/app.asar/build/main/main.js:142:932)
    at Module.Qc [as launch] (/nix/store/g87rxkfkfs11r0w7wmsrcdz9hbw1f6nw-heroic-unwrapped-2.18.1/opt/heroic/resources/app.asar/build/main/main.js:16:5680)
    at No (/nix/store/g87rxkfkfs11r0w7wmsrcdz9hbw1f6nw-heroic-unwrapped-2.18.1/opt/heroic/resources/app.asar/build/main/main.js:128:1469)
    at Session.<anonymous> (node:electron/js2c/browser_init:2:106823)
(14:54:26) [INFO]:    [Gog]:             Launching Comet!
(14:54:26) [INFO]:    [Gog]:             Launching Quake II: HEROIC_APP_NAME=1947927225 HEROIC_APP_RUNNER=gog GAMEID=umu-0 HEROIC_APP_SOURCE=gog STORE=gog STEAM_COMPAT_INSTALL_PATH="/home/keenan/Games/Heroic/Quake II Enhanced" PULSE_SINK=Game DXIL_SPIRV_CONFIG=wmma_rdna3_workaround LD_PRELOAD= STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/keenan/.local/share/.steam/steam WINEPREFIX="/home/keenan/Games/Heroic/Prefixes/default/Quake II" STEAM_COMPAT_DATA_PATH="/home/keenan/Games/Heroic/Prefixes/default/Quake II" PROTONPATH=/home/keenan/.config/heroic/tools/proton/proton-ge-bin WINE_FULLSCREEN_FSR=0 PROTON_ENABLE_WAYLAND=1 PROTON_ENABLE_HDR=1 PROTON_USE_WOW64=1 PROTON_DISABLE_NVAPI=1 PROTON_EAC_RUNTIME=/home/keenan/.config/heroic/tools/runtimes/eac_runtime PROTON_BATTLEYE_RUNTIME=/home/keenan/.config/heroic/tools/runtimes/battleye_runtime STEAM_COMPAT_APP_ID=0 SteamAppId=0 SteamGameId="heroic-Quake II Enhanced" PROTON_LOG_DIR=/home/keenan/.local/share GOGDL_CONFIG_PATH=/home/keenan/.config/heroic/gogdlConfig /nix/store/g87rxkfkfs11r0w7wmsrcdz9hbw1f6nw-heroic-unwrapped-2.18.1/opt/heroic/resources/app.asar.unpacked/build/bin/x64/linux/gogdl --auth-config-path /home/keenan/.config/heroic/gog_store/auth.json launch "/home/keenan/Games/Heroic/Quake II Enhanced" 1947927225 --no-wine --wrapper "obs-gamecapture /etc/profiles/per-user/keenan/bin/mangohud --dlsym "/usr/bin/umu-run"" --platform windows
(14:54:26) [DEBUG]:   [Gog]:             Logging to file(s) "/home/keenan/.local/state/Heroic/logs/games/1947927225_gog/launch.log", "/home/keenan/.local/state/Heroic/logs/runners/gog.log"
(14:54:26) [INFO]:    [Backend]:         Preventing display to sleep
(14:54:26) [INFO]:    [Gog]:             Got install info from cache for 1947927225 on windows platform
(14:54:26) [INFO]:    [Gog]:             Got install info from cache for 1947927225 on windows platform
(14:55:36) [INFO]:    [Gog]:             Killing Comet!
```

GalaxyCommunication.exe isn't in either prefix's `C:\ProgramData\GOG.com\Galaxy\redists`

## `nixpkgs-review` result for [#432183](https://github.com/NixOS/nixpkgs/pull/432183)

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 432183`

Logs: https://github.com/keenanweaver/nixpkgs-review-gha/actions/runs/17024754733


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>comet-gog_heroic</li>
    <li>heroic</li>
    <li>heroic-unwrapped</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>comet-gog_heroic</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>comet-gog_heroic</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>comet-gog_heroic</li>
  </ul>
</details>

aidalgol committed:
heroic-unwrapped: Apply patch for upstream bugfix PR
